### PR TITLE
Get storybook & typescript running

### PR DESCRIPTION
**Pull request by NickDunkman**
Merging into a feature branch! (`typeybook`)

- Removes a lot of the guts from our Fractal setup
- Adds storybook & typescript, with some basic configurations
- Ports over a single fractal page to a story (the main "overview" page)
- From what I've seen so far -- we'll probably want to have Storybook handle scss compilation so that sass changes can be hot-reloaded, so I'm also pulling down some of the extra css compilation scripts in package.json
- Storybook uses webpack internally, so I don't think we need to maintain a rollup configuration for our dev environment (a la `rollup.fractal-development.config.js`)

Screen Shot 2022-01-14 at 16 33 15

### CODE REVIEW

File: .storybook/theme.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { create } from '@storybook/theming';
+
+const logos = Object.freeze([
+  '/img/phenotypes-logo-green.svg',
+  '/img/phenotypes-logo-blue.svg',
+  '/img/phenotypes-logo-orange.svg',
+  '/img/phenotypes-logo-purple.svg',
+  '/img/phenotypes-logo-red.svg'
+]);
```
[the remark this answers is not in the document]
Comment (ahmedmodan) [reply .storybook/theme.js:9]: it also looks like when building static assets this image files are not being loaded properly. not sure if this is a known issue or not.

File: configs/rollup.phenotypes-library.config.js [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
-import babel from '@rollup/plugin-babel';
-import commonjs from '@rollup/plugin-commonjs';
-import resolve from '@rollup/plugin-node-resolve';
-import { WORKSPACE_ROOT } from './constants.js';
-import packageJSON from '../package.json';
+import peerDepsExternal from "rollup-plugin-peer-deps-external";
+import resolve from "@rollup/plugin-node-resolve";
+import commonjs from "@rollup/plugin-commonjs";
+import typescript from "rollup-plugin-typescript2";
+import postcss from "rollup-plugin-postcss";
```
Comment (ahmedmodan) [inline configs/rollup.phenotypes-library.config.js:7]: a js file but these have been auto-converted to double quotes instead of single? Is this file ignored by prettier?

File: configs/rollup.phenotypes-library.config.js [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
 
-const commonRollupConfig = {
+const packageJson = require("./package.json");
```
Comment (ahmedmodan) [inline configs/rollup.phenotypes-library.config.js:9]: why is this required instead of imported?

File: package.json [hunk 4]
```diff
@@ -41,23 +44,24 @@
...
     "test": "mocha --require @babel/register --require tests/test_setup.js --recursive library/components/**/*.spec.jsx",
     "buildCss": "node scripts/buildScss.js",
     "watchCss": "npm run buildCss && node scripts/watchScss.js",
-    "buildJs": "rollup --config ./configs/rollup.fractal-development.config.js",
-    "watchJs": "rollup --config ./configs/rollup.fractal-development.config.js --watch",
-    "buildAll": "npm run buildJs && npm run buildCss",
-    "watchAll": "npm run watchCss & npm run watchJs",
-    "startProd": "npm run buildAll & fractal start",
-    "start": "fractal start --sync & npm run watchAll",
+    "startProd": "npm run buildCss && npm run storybook",
```
Comment (ahmedmodan) [inline package.json:56]: Do we want to change how this is deployed? We can deploy just the static assets if we want instead of running the server in prod. If that is the case this should be changed to `npm run buildCss && npm run build-storybook`

**Review by ahmedmodan — COMMENTED** [review]
looks pretty good so far a few questions mainly.

File: .storybook/main.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  "stories": [
+    "../guides/**/*.stories.@(md|mdx)",
+    //"../src/**/*.stories.@(js|jsx|ts|tsx|md|mdx)"
```
[the remark this answers is not in the document]
Comment (NickDunkman) [reply .storybook/main.js:4]: This is where the stories for components/utilities/etc will live -- am thinking of this directory structure:

```
phenotypes/
    .storybook/ -> anything that is specific to storybook goes in here, hidden from everything else which makes phenotypes, phenotypes
    dist/ -> compiled stuff for npm
    guides/ -> the guides -- not in /src b/c it does not contain code that contributes to the published package
    src/ -> contains all code that is published, + code that needs to be co-located (stories, unit tests)
        base/ -> the sass files that pull everything together for phenotypes.css
        components/
        utilities/
```

File: .storybook/main.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  ],
+  "addons": [
+    "@storybook/addon-links",
+    "@storybook/addon-essentials"
+  ],
+  "staticDirs": ["./public"],
+  "framework": "@storybook/react"
+}
```
[the remark this answers is not in the document]
Comment (NickDunkman) [reply .storybook/main.js:12]: 👍 will configure that in next PR

File: configs/rollup.phenotypes-library.config.js (hunk 2 above)
Comment (NickDunkman) [reply configs/rollup.phenotypes-library.config.js]: Fixed! https://github.com/aminohealth/phenotypes/pull/168/commits/a83c1c48356d556282fbdaf15c37f6dcd8ff4903

File: configs/rollup.phenotypes-library.config.js (hunk 3 above)
Comment (NickDunkman) [reply configs/rollup.phenotypes-library.config.js]: Ya not sure -- I left this as is -- I'll add this to my mental todo list to convert to an import when circling back to publishing stuff later in the project, I can't remember how `import`-ing json works off the top of my head (whether is default import or not)

File: .storybook/theme.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (NickDunkman) [reply .storybook/theme.js]: Yea it's overkill to freeze that, removed here: https://github.com/aminohealth/phenotypes/pull/168/commits/274fe8fd79d30b31af1ed8c9a611930138267e42

I have a separate task to work on the static stuff for prod later on, I'm sure the logo stuff will stick out again at that point.

File: package.json (hunk 4 above)
Comment (NickDunkman) [reply package.json:56]: You yea definitely -- I put this as a pre-MVP for now, will work on static build stuff separately later.

File: configs/rollup.phenotypes-library.config.js (hunk 3 above)
Comment (ahmedmodan) [reply configs/rollup.phenotypes-library.config.js]: this was previously imported, see line 7 in the previous diff, It works as a default import. Let's update it to make it consistent.

Comment (NickDunkman) [reply configs/rollup.phenotypes-library.config.js]: Oh sorry, I read the diff wrong! Updating ...

Comment (NickDunkman) [reply configs/rollup.phenotypes-library.config.js]: Oof I really clobbered the imports in that file -- fixed: https://github.com/aminohealth/phenotypes/pull/168/commits/060dea25405b216d04dcfa81f0afc2eb85b253ca

**Review by ahmedmodan — APPROVED** [review]
